docs/proposals/volumes.md

```diff
@@ -0,0 +1,431 @@
+as local volumes.  Since inodes are written to these block devices in the same way as local devices,
+permissions and ownership can be managed on the client side by the Kubelet.
+
+#### Shared storage: hostPath
+
+The `hostPath` volume should only be used by 'trusted' users, and the permissions of paths exposed
+into containers via hostPath volumes should always be managed by the cluster operator.  If the
+Kubelet managed the ownership for `hostPath` volumes, a user who could create a `hostPath`
+volume could affect changes in the state of arbitrary paths within the host's filesystem.  This
+would be a severe security risk, so we will consider hostPath a corner case that the kubelet should
+never perform ownership management for.
+
+#### Shared storage: network
+
+Ownership management of shared storage is a complex topic.  Ownership for existing shared storage
+will be managed externally from Kubernetes.  For this case, our API should make it simple to express
+whether a particular volume should have these concerns managed by Kubernetes.
+
+We will not attempt to address the actual ownership and permissions concerns of new shared storage
+in this proposal.
+
+#### API requirements
+
+From the above, we know that ownership management must be applied:
+
+1.  To some volume types always
+2.  To some volume types never
+3.  To some volume types *sometimes*
+
+For volume types where the need for ownership/label management is not at the discretion of the user,
+there should be no need (or way) to specify whether ownership management is required in the API.
+Conversely there must be a way to specify this for volume types where it depends on the user's
+needs.
+
+### Ownership management
+
+Using the approach of a pod-level supplemental group to own volumes solves the problem in any of the
+cases of UID/GID combinations within a pod. Since this is the simplest approach that handles all
+use-cases, our solution will be made in terms of it.
+
+Eventually, Kubernetes should allocate a unique group for each pod so that a pod's volumes are
+usable by that pod's containers, but not by containers of another pod.  The supplemental group used
+to share volumes must be unique in a multitenant cluster.  If uniqueness is enforced at the host
+level, pods from one host may be able to use shared filesystems meant for pods on another host.
+
+An admission controller could handle allocating groups for each pod and setting the group in the
+pod's security context.
+
+#### When to manage ownership
+
+Whether Kubernetes should manage the ownership of a volume for a volume depends upon
+both the volume type and the cluster operator's policy.  For example:
+
+1.  For shared storage, ownership of volumes externally to the cluster
+2.  It is not possible to securely `chown` or `chmod` paths within some shared storage
+
+The system should be designed so that ownership management is only applied to the appropriate
+volumes.
+
+### Setting ownership and permissions on volumes
```

Comment:
This paragraph seems like it would be better placed above - "For local volumes and unshared storage, chown and chmod are sufficient locally to assign membership.  For other volumes, XXX must be done".  As-is these three sections (Ownership Management, when to manage, and setting) don't flow cleanly.